### PR TITLE
Make 45

### PR DESCRIPTION
**Pull request by GeorgeKerwood**
Three commits improving curve repointing.

### CODE REVIEW

**Comment by GeorgeKerwood** [comment]
@lukeedw New commits resolving tab issues.

File: Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/Wireframe Tests/Generic Curve Tests/CurveTest.cs [hunk 1]
```diff
@@ -536,9 +536,18 @@ public void RepointCurveTest()
...
-        /// A test for Reverse
+        /// A test for toleranced Repoint
         /// </summary>
         [Test]
+        public void RepointTolerancedCurveTest()
+        {
+            RepointTolerancedTest(TestFiles.SINGLE_CURVE);
+        }
+
+		/// <summary>
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/Wireframe Tests/Generic Curve Tests/CurveTest.cs]: Odd tabbing here too.

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs [hunk 2]
```diff
@@ -216,6 +216,14 @@ public void DeleteCurvePoints(int[] pointsIndeces)
             {
                 throw new Exception("Number of points must be 2 or more");
             }
+            if (startPoint > endPoint)
+            {
+                throw new Exception("Start Point must proceed End Point");
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs]: "preceed" not "proceed"

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs [hunk 3]
```diff
@@ -216,6 +216,14 @@ public void DeleteCurvePoints(int[] pointsIndeces)
...
+            }
+            if (startPoint == endPoint & !IsClosed)
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs:223]: How can they be the same if the curve is closed? Not sure how that works?

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs [hunk 4]
```diff
@@ -231,13 +239,75 @@ public void DeleteCurvePoints(int[] pointsIndeces)
...
+
+            if (startPoint < 1 || startPoint > NumberPoints)
+            {
+                throw new Exception("Start Point is not defined");
+            }
+            if (endPoint < 1 || endPoint > NumberPoints)
+            {
+                throw new Exception("End Point is not defined");
+            }
+            if (tolerance < 1e-5)
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs:276]: Is this the right tolerance or should you compare it to the general tolerance in PowerShape?

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs [hunk 5]
```diff
@@ -231,13 +239,75 @@ public void DeleteCurvePoints(int[] pointsIndeces)
...
+            {
+                throw new Exception("Tolerance must be greater or equal to 1e-5");
+            }
+            if (startPoint > endPoint)
+            {
+                throw new Exception("Start Point must proceed End Point");
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs]: "Preceed"

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs [hunk 6]
```diff
@@ -231,13 +239,75 @@ public void DeleteCurvePoints(int[] pointsIndeces)
...
+            }
+            if (startPoint == endPoint & !IsClosed)
```
Comment (lukeedw) [inline Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs:284]: Same comment as before, how can they be the same point?

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs (hunk 4 above)
Comment (GeorgeKerwood) [reply Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs:276]: This tolerance can be specified independently of the session's general tolerance. The minimum value supported is 1e-5.

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs (hunk 3 above)
Comment (GeorgeKerwood) [reply Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs:223]: Yeah, granted it's weird: It's span based, you're specifying the repointing to occur in a range. For example if I have a 5 point closed curve, I have 5 spans: 1-2, 2-3, 3-4, 4-5, 5-1. The whole curve is therefore the span 1-1, or any other index that starts and ends at the same place.

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs (hunk 6 above)
Comment (GeorgeKerwood) [reply Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs:284]: As above

File: Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs (hunk 5 above)
Comment (GeorgeKerwood) [reply Delcam.ProductInterface.PowerShape/Entities/Wireframe/PSGenericCurve.cs]: Close :P "Precede"

**Pull request merged**